src/components/structures/RoomView.js

```diff
@@ -782,32 +792,51 @@ export default createReactClass({
                     e2eStatus: hasUnverifiedDevices ? "warning" : "verified",
                 });
             });
+            debuglog("e2e check is warning/verified only as cross-signing is off");
             return;
         }
+
+        /* At this point, the user has encryption on and cross-signing on */
         const e2eMembers = await room.getEncryptionTargetMembers();
-        for (const member of e2eMembers) {
-            const { userId } = member;
-            const userVerified = cli.checkUserTrust(userId).isCrossSigningVerified();
-            if (!userVerified) {
-                this.setState({
-                    e2eStatus: "warning",
-                });
-                return;
-            }
+        const verified = [];
+        const unverified = [];
+        e2eMembers.map(({userId}) => userId)
+            .filter((userId) => userId !== cli.getUserId())
+            .forEach((userId) => {
+                (cli.checkUserTrust(userId).isCrossSigningVerified() ?
+                verified : unverified).push(userId)
+            });
+
+        debuglog("e2e verified", verified, "unverified", unverified);
+
+        /* If we verify any users in this room, expect to verify every user in the room */
+        if (verified.length > 0 && unverified.length > 0) {
+            this.setState({
+                e2eStatus: "warning",
+            });
+            debuglog("e2e status set to warning as some, but not all, users are verified");
+            return;
+        }
+
+        /* At this point, either `verified` or `unverified` is empty, or both */
+        /* Check all verified user devices. We don't care if everyone's unverified anyway. */
+        for (const userId of [...verified, cli.getUserId()]) {
```

Comment:
I don't think we should ever be considering ourselves when calculating whether the room is trusted, since that would imply all rooms become untrusted if you sign in and don't verify that sign in, which is probably not the right way to convey that badness.

Reply:
done